### PR TITLE
[Workspace] Add update workspace page

### PR DESCRIPTION
**Pull request by gaobinlong**
### Description

This PR is followed by https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6179, which adds the update workspace page that can be used to update the workspace's attributes such as name, description and feature set etc.

For the implementation, this new page leverages most of the existing components in the create workspace page.

### Issues Resolved

https://github.com/opensearch-project/OpenSearch-Dashboards/issues/6269

## Screenshot

image

https://github.com/opensearch-project/OpenSearch-Dashboards/assets/15811131/718babf8-fdf8-479b-bce4-0e96928cd16b

## Testing the changes

1. Clone the latest code and run yarn osd bootstrap
2. Modify `config/opensearch_dashboards.yml`, add `workspace.enabled: true`
3. Run `yarn start --no-base-path`
4. Create a workspace firstly, or click `Select a workspace` to enter the overview page of a workspace, the URL is like this: `http://localhost:5601/w/DsNfKS/app/workspace_overview`
5. Visit the URL `http://localhost:5601/w/DsNfKS/app/workspace_update`.
6. Update the name, description, feature set or other attributes for the workspace.

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [x] New functionality includes testing.
- [x] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by gaobinlong** [comment]
Hi @Flyingliuhub, could you help to review this PR, thank you~

File: src/plugins/workspace/public/components/workspace_updater/workspace_updater.tsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        }
+        return;
+      }
+      notifications?.toasts.addDanger({
+        title: i18n.translate('workspace.update.failed', {
+          defaultMessage: 'Failed to update workspace',
+        }),
+        text: result?.error,
+      });
+    },
```
Comment (xinruiba) [inline src/plugins/workspace/public/components/workspace_updater/workspace_updater.tsx:89]: Thanks for this change. Looks like the error handling is duplicate, what if we change it like this:

```suggestion
      try {
        const { ...attributes } = data;
        result = await workspaceClient.update(currentWorkspace.id, attributes);
        if (result?.success) {
          notifications?.toasts.addSuccess({
            title: i18n.translate('workspace.update.success', {
              defaultMessage: 'Update workspace successfully',
            }),
          });
          if (application && http) {
            // Redirect page after one second, leave one second time to show update successful toast.
            window.setTimeout(() => {
              window.location.href = formatUrlWithWorkspaceId(
                application.getUrlForApp(WORKSPACE_OVERVIEW_APP_ID, {
                  absolute: true,
                }),
                currentWorkspace.id,
                http.basePath
              );
            }, 1000);
          }
          return;
        } else {
          throw Error('workspace.update.success failed')
        }
      } catch (error) {
        notifications?.toasts.addDanger({
          title: i18n.translate('workspace.update.failed', {
            defaultMessage: 'Failed to update workspace',
          }),
          text: error instanceof Error ? error.message : JSON.stringify(error),
        });
        return;
      }
```

The disadvantage in the suggestion change is, we also include ```window.setTimeout()``` and other window API call into the error handling, which for me it's ok :)

What do you think?

Comment (gaobinlong) [reply src/plugins/workspace/public/components/workspace_updater/workspace_updater.tsx]: Thanks for your review, the suggested change looks good to me, and I've changed the code in both update workspace page and create workspace page, please help to take a look.

**Comment by xinruiba** [comment]
LGTM, thanks

**Review by xinruiba — APPROVED** [review]

**Review by SuZhou-Joe — APPROVED** [review]

**Pull request merged**